src/exchanges/bitfinex-client.js

```diff
@@ -238,21 +267,37 @@ class BitfinexClient extends BasicClient {
 
     let isDelete = count === 0;
     if (isDelete) point.size = (0).toFixed(8); // reset the size to 0, comes in as 1 or -1 to indicate bid/ask
-
     let update = new Level2Update({
       exchange: "Bitfinex",
       base: market.base,
       quote: market.quote,
+      sequenceId: +sequence,
       asks,
       bids,
     });
     this.emit("l2update", update, market);
   }
 
   _onLevel3Snapshot(msg, market) {
+    /*
+    example msg:
+    [
+      648087,
+      [ 
+        [ 55888179267, 31111, 0.05 ],
+        [ 55895806791, 31111, 0.989 ],
+        ...
+      ],
+      1
+    ]
+    */
     let bids = [];
     let asks = [];
-    for (let p of msg[1]) {
+    
+    let orders = msg[1];
+    let sequence = +msg[2];
+
+    for (let p of orders) {
```

Comment:
nit: maybe destructure to `[orderId, price, size]` to be more readable instead of carrying `p` forward